tests/integration/src/test/java/org/mule/config/spring/SpringRegistryConcurrencyTestCase.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.config.spring;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.junit.Assert.assertThat;
+import org.mule.api.MuleException;
+import org.mule.api.lifecycle.Disposable;
+import org.mule.api.lifecycle.Startable;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.testmodels.fruit.Apple;
+import org.mule.util.concurrent.Latch;
+
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+import org.springframework.beans.factory.FactoryBean;
+
+public class SpringRegistryConcurrencyTestCase extends FunctionalTestCase
+{
+
+    private static final String KEY = "KEY";
+    private static final long LATCH_TIMEOUT = 5;
+
+    private Latch latch;
+    private Latch auxiliaryLatch;
+    private Map<String, Object> lookupObjects;
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "spring-registry-concurrency-config.xml";
+    }
+
+    @Override
+    protected void doSetUp() throws Exception
+    {
+        latch = new Latch();
+        auxiliaryLatch = new Latch();
+        lookupObjects = null;
```

Comment:
might not be strictly required. But it's a good practice to clear state completely since the tests do write to that field